### PR TITLE
Fix documentation in cell mod

### PR DESCRIPTION
**Pull request by F001**
The implementation of Rc type in this doc is outdated.

### CODE REVIEW

File: src/libcore/cell.rs [hunk 1]
```diff
@@ -121,22 +121,26 @@
 //! ```
 //! use std::cell::Cell;
 //!
-//! struct Rc<T> {
-//!     ptr: *mut RcBox<T>
+//! pub struct Rc<T: ?Sized> {
```
Comment (GuillaumeGomez) [inline src/libcore/cell.rs]: Why adding `pub` in here?

**Comment by steveklabnik** [comment]
Seems like the build failure is spurious.

I share @GuillaumeGomez 's question about `pub`, but otherwise looks good to me

**Comment by GuillaumeGomez** [comment]
Travis failed. Run:

```bash
make tidy
```

to see what's wrong.

File: src/libcore/cell.rs (hunk 1 above)
Comment (F001) [inline src/libcore/cell.rs]: Agree. The pub keyword is removed.

**Comment by steveklabnik** [comment]
Hm, still a strange failure. Let's see how it does on real CI, thanks!

@bors: r+ rollup

File: src/libcore/cell.rs [hunk 2]
```diff
@@ -121,22 +121,26 @@
...
-//!             Rc { ptr: self.ptr }
-//!         }
+//!         self.inc_strong();
+//!         Rc { ptr: self.ptr }
+//!     }
+//! }
+//!
+//! trait RcBoxPtr<T: ?Sized> {
+//!     fn inc_strong(&self) {
+//!         self.inner().strong.set(self.strong().checked_add(1).unwrap_or_else(|| unsafe { abort() }));
```
Comment (eddyb) [inline src/libcore/cell.rs]: This is still over 100 columns wide, please split into multiple lines.

**Comment by GuillaumeGomez** [comment]
And run `make tidy`.

**Comment by GuillaumeGomez** [comment]
And please squash your commits.

**Comment by F001** [comment]
@steveklabnik  Just Ping. Could you please help to merge this PR?

**Pull request merged**